app/Http/Controllers/AdminDash.php

```diff
@@ -1370,15 +1371,35 @@ public function saveNewEvent(Request $request) {
             'end_time' => 'required',
             'description' => 'required'
         ]);
-
-        if ($request->file('banner') != null) {
+        
+        if ($request->file('banner') != null and $request->filled('banner_url')) {
+            return redirect()->back()->withErrors('Please ensure you submit only one of the following: a URL or a file for the banner.')->withInput();
+        } elseif ($request->file('banner') != null) {
             $ext = $request->file('banner')->getClientOriginalExtension();
             $time = Carbon::now()->timestamp;
             $path = $request->file('banner')->storeAs(
                 'public/event_banners',
                 $time.'.'.$ext
             );
             $public_url = '/storage/event_banners/'.$time.'.'.$ext;
+        } elseif ($request->filled('banner_url')) {
+            try {
+                $response = Http::get($request->banner_url);
+                if (!$response->successful()) {
+                    return redirect()->back()->withErrors('The provided URL cannot be accessed')->withInput();
+                }
+            } catch (\Exception $e) {
+                return redirect()->back()->withErrors('The provided URL is not valid')->withInput();
+            }
```

Comment:
Is there a way we can throw an exception if `! $response->successful()`? That way we only need one return here in the catch